horizon/facts/opal_forwarder.py

```diff
@@ -59,10 +64,26 @@ def create_data_source_entry(
     )
 
 
-def create_data_update_entry(entries: list[DataSourceEntry]) -> DataUpdate:
+def create_data_update_entry(
+    entries: list[DataSourceEntry],
+    *,
+    update_id: UUID | None = None,
+) -> DataUpdate:
     entries_text = ", ".join(entry.dst_path for entry in entries)
+    _update_id = (update_id or uuid4()).hex
+
+    if update_id is None:
+
+        def inject_update_id(entry: DataSourceEntry) -> DataSourceEntry:
+            entry_headers = entry.config.get("headers", {})
+            entry_headers["X-Permit-Update-Id"] = _update_id
+            entry.config["headers"] = entry_headers
+            return entry
+
+        entries = list(map(inject_update_id, entries))
```

Comment:
Why is this still here?

Reply:
In case update id was not provided i want to inject it